platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/editor/StateEditor.java

```diff
@@ -42,29 +48,34 @@
  */
 public class StateEditor {
 
-    private SignedState signedState;
+    private ReservedSignedState signedState;
     private MerkleRoute currentWorkingRoute = MerkleRouteFactory.getEmptyRoute();
     private boolean alive = true;
+    private final PlatformContext platformContext;
 
     /**
      * Create a new state editor.
      *
-     * @param statePath
-     * 		the path where the signed state can be found
+     * @param statePath the path where the signed state can be found
      */
     public StateEditor(final Path statePath) throws IOException {
-        final DeserializedSignedState deserializedSignedState = SignedStateFileReader.readStateFile(statePath);
+
+        platformContext = new DefaultPlatformContext(
+                ConfigurationBuilder.create().build(), new NoOpMetrics(), CryptographyHolder.get());
+
+        final DeserializedSignedState deserializedSignedState =
+                SignedStateFileReader.readStateFile(platformContext, statePath);
         System.out.println("\nLoading state from " + statePath);
-        signedState = deserializedSignedState.signedState();
+        signedState = deserializedSignedState.reservedSignedState();
```

Comment:
I dont see this being closed anywhere. I know this is just a utility for PCLI, but still good to do this properly to reduce the chance of a reference count bug to sneak in where it matters. For example, if someone looks at this code as an example for core platform code.

Reply:
This one actually is reserved, but it's a bit inelegant the way it's done. I'm going to migrate this to using a `SignedStateReference`, as this is the recommended strategy in my list of rules for holding a state in a class like this.